arxiv/base/templates/base/header.html

```diff
@@ -13,6 +13,24 @@ <h1 class="level-item"><a href="{{ url_for('home') }}" aria-label="arxiv-logo">a
   {{ macros.compactsearch('level-right') }}
 </div> <!-- closes identity -->
 
-<!--  # TODO: reintroduce this once we have access to the user's session.
-  <div class="user-tools box is-pulled-right" role="navigation" aria-label="User menu"><a href="{{ url_for('login') }}">Login</a> | <a href="{{ url_for('account') }}">My Account</a> | <a href="{{ url_for('logout') }}">Logout</a></div>
--->
+<div class="is-pulled-right">
+  <div class="user-tools box level-item" role="navigation" aria-label="User menu">
+    {% if request.session and request.session.user %}Logged in as <strong>{{ request.session.user.username }}</strong> | <a href="{{ url_for('account') }}">My Account</a> | <a href="{{ url_for('logout') }}">Logout</a>{% else %}<a href="{{ url_for('login') }}">Login</a>{% endif %}
+  </div>
+</div>
+
+<div class="container is-fluid notifications">
```

Comment:
I would suggest having the notification width be set at this level -- to avoid possibly making all notifications in other contexts an undesired width.
Also: add `role=alert` to this div to have screen readers interrupt flow of document to bring flash messages first.

Reply:
I'm having trouble getting this to work at the container level. If I use ``!important`` when setting width at that level, I am able to get the desired width. But because of the way centering works with the container it gets pulled offcenter to the left. Tempted to leave this for now, and you may have better luck?